packages/backend/src/api/controllers/tvl/TvlController.ts

```diff
@@ -100,6 +115,59 @@ export class TvlController {
     }
   }
 
+  async getAggregatedApiResponse(slugs: string[]): Promise<TvlProjectResult> {
+    const dataTimings = await this.getDataTimings()
+
+    const projectIdsFilter = [...layer2s, ...bridges]
+      .filter((project) => slugs.includes(project.display.slug))
+      .map((project) => project.id)
+
+    if (!dataTimings.latestTimestamp) {
+      return {
+        result: 'error',
+        error: 'NO_DATA',
+      }
+    }
+
+    if (!dataTimings.isSynced && this.options.errorOnUnsyncedTvl) {
+      return {
+        result: 'error',
+        error: 'DATA_NOT_FULLY_SYNCED',
+      }
+    }
+
+    const [hourlyReports, sixHourlyReports, dailyReports] = await Promise.all([
+      this.aggregatedReportRepository.getHourly(
+        getHourlyMinTimestamp(dataTimings.latestTimestamp),
+        'TVL',
+      ),
+      this.aggregatedReportRepository.getSixHourly(
+        getSixHourlyMinTimestamp(dataTimings.latestTimestamp),
+        'TVL',
+      ),
+      this.aggregatedReportRepository.getDaily('TVL'),
+    ])
+
+    const projectIdsFilterSet = new Set(projectIdsFilter)
```

Comment:
Why do we need the Set, can't we just use the array?

Reply:
The Set has better performance, so I picked that.
I also fixed an issue, we were comparing references (projectId is not string)